ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala

```diff
@@ -146,26 +149,25 @@ class H2OGridSearch(override val uid: String)
     if (unsortedGridModels.isEmpty) {
       throw new IllegalArgumentException("No model returned.")
     }
-    gridModels = sortGridModels(unsortedGridModels)
+    gridModels = sortGridModels(unsortedGridModels, algoName)
     gridModels.head
   }
 
-  private def sortGridModels(gridModels: Array[H2OMOJOModel]): Array[H2OMOJOModel] = {
+  private def sortGridModels(gridModels: Array[H2OMOJOModel], algoName: String): Array[H2OMOJOModel] = {
     val metric = if (getSelectBestModelBy() == H2OMetric.AUTO.name()) {
       val category = H2OModelCategory.fromString(gridModels(0).getModelCategory())
       category match {
         case H2OModelCategory.Regression => H2OMetric.RMSE
         case H2OModelCategory.Binomial => H2OMetric.AUC
         case H2OModelCategory.Multinomial => H2OMetric.Logloss
         case H2OModelCategory.Clustering => H2OMetric.TotWithinss
+        case H2OModelCategory.DimReduction if algoName == "glrm" => H2OMetric.GLRMMetric
```

Comment:
NIT: `if algoName == "glrm"` is required?

Reply:
For now, I can remove it. But in future with more algorithms, there will have to be such a condition since the metric is specific only to GLRM.

Reply:
It is fine, we can keep this there, was just asking :)